app/design-system/design-system-hedvig/src/main/kotlin/com/hedvig/android/design/system/hedvig/RadioOption.kt

```diff
@@ -0,0 +1,562 @@
+                tint = Color.Unspecified,
+                modifier = Modifier
+                  .size(32.dp),
+              )
+            }
+
+            is IconResource.Painter -> {
+              Image(
+                painter = painterResource(id = radioOptionStyle.iconResource.painterResId),
+                contentDescription = null,
+                modifier = Modifier
+                  .size(32.dp),
+              )
+            }
+          }
+          Spacer(Modifier.width(8.dp))
+          HedvigText(
+            optionText,
+            style = radioOptionSize.size(radioOptionStyle).optionTextStyle,
+            modifier = Modifier.weight(1f),
+            color = optionTextColor,
+          )
+          Spacer(Modifier.width(8.dp))
+          SelectIndicationCircle(chosenState, lockedState)
+        }
+
+        is Label -> {
+          Column(Modifier.weight(1f)) {
+            HedvigText(
+              optionText,
+              style = radioOptionSize.size(radioOptionStyle).optionTextStyle,
+              color = optionTextColor,
+            )
+            HedvigText(
+              radioOptionStyle.labelText,
+              style = radioOptionSize.size(radioOptionStyle).labelTextStyle,
+              color = labelTextColor,
+            )
+          }
+          Spacer(Modifier.width(8.dp))
+          SelectIndicationCircle(chosenState, lockedState)
+        }
+
+        LeftAligned -> {
+          SelectIndicationCircle(chosenState, lockedState)
+          Spacer(Modifier.width(8.dp))
+          HedvigText(
+            optionText,
+            style = radioOptionSize.size(radioOptionStyle).optionTextStyle,
+            color = optionTextColor,
+            modifier = Modifier.weight(1f),
+          )
+        }
+      }
+    }
+  }
+}
+
+@Composable
+internal fun SelectIndicationCircle(
+  chosenState: ChosenState,
+  lockedState: LockedState,
+  modifier: Modifier = Modifier,
+) {
+  CheckItemAnimation(chosenState) { currentState: ChosenState ->
+    Box(
+      contentAlignment = Alignment.Center,
+    ) {
+      Spacer(
+        modifier
+          .size(24.dp)
+          .clip(CircleShape)
+          .then(
+            when (currentState) {
+              Chosen -> {
+                when (lockedState) {
+                  Locked -> Modifier.border(8.dp, radioOptionColors.disabledIndicatorColor, CircleShape)
+                  NotLocked -> Modifier.border(8.dp, radioOptionColors.chosenIndicatorColor, CircleShape)
+                }
+              }
+
+              NotChosen -> {
+                Modifier.border(2.dp, radioOptionColors.notChosenIndicatorColor, CircleShape)
+                // disabled color is actually brighter than notChosenIndicatorColor,
+                // makes no sense here, so I left only one
+              }
+            },
+          ),
+      )
+    }
+  }
+}
+
+@Composable
+internal fun CheckItemAnimation(state: ChosenState, content: @Composable (state: ChosenState) -> Unit) {
```

Comment:
Is this internal function used in other files too or could it be private?

Reply:
Yep, it is used in checkbox too